pkg/mdformatter/linktransformer/config.go

```diff
@@ -23,24 +23,30 @@ type Config struct {
 }
 
 type ValidatorConfig struct {
-	// Regex for type github is reponame matcher, like `bwplotka\/mdox`.
+	// Regex for type of validator. For `githubPullsIssues` this is: (^http[s]?:\/\/)(www\.)?(github\.com\/){ORG_NAME}\/{REPO_NAME}(\/pull\/|\/issues\/).
 	Regex string `yaml:"regex"`
-	// By default type is `ignore`. Could be `github` or `roundtrip`.
+	// By default type is `roundtrip`. Could be `githubPullsIssues` or `ignore`.
 	Type ValidatorType `yaml:"type"`
 	// GitHub repo token to avoid getting rate limited.
 	Token string `yaml:"token"`
+	// Validators to be used alongside githubPullsIssues.
+	NPIValidators []NonPullsIssuesValidator `yaml:"nonPullsIssuesValidators"`
```

Comment:
hmmm, I am not fan of this approach...

Reasons:
* We are leaking GH special flow. It's actually quite surprise to guess what is the logic here
* If you regex only fo pulls or issues how other links even get there? In your example I would expect only pulls and issues being matched here so nonXYZ thing will never get called (at least that's my perception)

```
  - regex: '(^http[s]?:\/\/)(www\.)?(gitlab\.com\/)thanos-io\/thanos(\/pull\/|\/issues\/)'
    type: 'githubPullsIssues'
    nonPullsIssuesValidators:
      - regex: '(^http[s]?:\/\/)(www\.)?(github\.com\/)thanos-io\/thanos(\/releases\/tag\/)'
      - regex: '(^http[s]?:\/\/)(www\.)?(github\.com\/)thanos-io\/thanos(\/blob\/)'
      - regex: '(^http[s]?:\/\/)(www\.)?(github\.com\/)thanos-io\/thanos(\/commit\/)'```
```

Do we need this. Why not simple approach of regexing pull and issues only (and erroring out when user passes non issue or non PR) then use ignore type to ignore further stuff? 🤔

Reply:
Yes, I was kind of mixing nested validators + `pullsIssues` which on second thought might make things super confusing 😅. `ignore` would be sufficient for the `nonPullsIssues` thing. Will remove this and just have `githubPullsIssues`. So example config would be,
```
  - regex: '(^http[s]?:\/\/)(www\.)?(gitlab\.com\/)thanos-io\/thanos(\/pull\/|\/issues\/)'
    type: 'githubPullsIssues'
  - regex: '(^http[s]?:\/\/)(www\.)?(github\.com\/)thanos-io\/thanos(\/releases\/tag\/)'
    type: 'ignore'
```